incubator/virtualcluster/cmd/syncer/app/options/options.go

```diff
@@ -156,7 +164,47 @@ func (o *ResourceSyncerOptions) Config() (*syncerappconfig.Config, error) {
 	c.ComponentConfig = o.ComponentConfig
 
 	// Prepare kube clients
-	leaderElectionClient, virtualClusterClient, superMasterClient, restConfig, err := createClients(c.ComponentConfig.ClientConnection, o.SuperMaster, c.ComponentConfig.LeaderElection.RenewDeadline.Duration)
+	var (
+		metaRestConfig, restConfig, inClusterConfig *restclient.Config
+		err                                         error
+	)
+	restConfig, err = getClientConfig(c.ComponentConfig.ClientConnection, o.SuperClusterAddress, !o.DeployOnMetaCluster)
```

Comment:
How about renaming restConfig to superRestConfig? 
How about removing inClusterConfig and use leaderElectionRestConfig directly?

Reply:
fixed